src/sentry/integrations/base.py

```diff
@@ -123,11 +123,6 @@ class is just a descriptor for how that object functions, and what behavior
     it provides (such as extensions provided).
     """
 
-    # a unique identifier (e.g. 'slack').
-    # Used to lookup sibling classes and the ``key`` used when creating
-    # Integration objects.
-    key: Optional[str] = None
```

Comment:
Why did you remove this?

Reply:
I think this will cause an error if a subclass hasn't implemented `key`. If it's required we might want to make it an abstract property.

Reply:
I tried moving `key` to `PipelineProvider` as an abstract class property so that `IntegrationProvider` and others could use it without having to redefine it. 
![Screen Shot 2022-04-04 at 4 52 38 PM](https://user-images.githubusercontent.com/31750075/161651026-40217b9d-d7a1-475e-9dda-319bd9bbd246.png)
I'm going to revert because I think a refactor like that should happen in python 3.10 when we have a better decorator.

Reply:
It turns out that I actually really do have to add `key` and `name` to `PipelineProvider` so I've pulled that into a separate PR: https://github.com/getsentry/sentry/pull/33348.